main.tf

```diff
@@ -117,18 +106,10 @@ resource "azurerm_route_table" "rt" {
       next_hop_in_ip_address = lookup(route.value, "next_hop_in_ip_address", null)
     }
   }
-  disable_bgp_route_propagation = var.disable_bgp_route_propagation
-  tags                          = module.labels.tags
 }
 
 resource "azurerm_subnet_route_table_association" "main" {
-  count          = var.enable && var.enable_route_table && var.specific_name_subnet == false ? length(var.subnet_prefixes) : 0
-  subnet_id      = element(azurerm_subnet.subnet.*.id, count.index)
-  route_table_id = join("", azurerm_route_table.rt.*.id)
-}
-
-resource "azurerm_subnet_route_table_association" "main2" {
-  count          = var.enable && var.enable_route_table && var.specific_name_subnet ? length(var.subnet_prefixes) : 0
-  subnet_id      = element(azurerm_subnet.subnet2.*.id, count.index)
-  route_table_id = join("", azurerm_route_table.rt.*.id)
+  count          = var.enable ? local.subnet : 0
```

Comment:
Add enable `var.enable_route_table` variable also. Because when we don't want to create route table than this resource should also not get created.